### PR TITLE
[W06H01] Rewrote existing tests a bit to cover more details

### PR DESCRIPTION
**Pull request by TheAleWu**
Updated the existing tests to cover more details like the prev and next element in the chain and to pre-compare the list sizes for premature test failure on inconsistency

### CODE REVIEW

**Comment by TheAleWu** [comment]
Ich bitte um ein Review

File: w06h01/test/pgdp/CountTreshTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            arguments(list(2), 2, new long[]{0, 2, 0}),
+            arguments(list(2), 3, new long[]{2, 0, 0}),
+            arguments(list(1, 2, 3, Integer.MIN_VALUE), Integer.MIN_VALUE, new long[]{0, Integer.MIN_VALUE, 6}),
+            arguments(list(1, 2, 3, Integer.MAX_VALUE), Integer.MAX_VALUE, new long[]{6, Integer.MAX_VALUE, 0}),
+            // "Die Methode kann auf jeder möglichen Liste aufgerufen werden, aber es kommt in unseren Tests nicht zu long-Overflows."
+            // => Die Methode muss auch mit Int Overflows umgehen können!
+            arguments(list(Integer.MAX_VALUE, Integer.MAX_VALUE, Integer.MAX_VALUE), Integer.MAX_VALUE, new long[]{0, Integer.MAX_VALUE * 3L, 0}),
+            arguments(list(1, 2, 3, Integer.MAX_VALUE, Integer.MAX_VALUE, Integer.MAX_VALUE), Integer.MAX_VALUE, new long[]{6, Integer.MAX_VALUE * 3L, 0}),
+            arguments(list(Integer.MIN_VALUE, Integer.MIN_VALUE, Integer.MIN_VALUE), Integer.MIN_VALUE, new long[]{0, Integer.MIN_VALUE * 3L, 0}),
+            arguments(list(1, 2, 3, Integer.MIN_VALUE, Integer.MIN_VALUE, Integer.MIN_VALUE), Integer.MIN_VALUE, new long[]{0, Integer.MIN_VALUE * 3L, 6})
```
Comment (nilsreichardt) [inline w06h01/test/pgdp/CountTreshTest.java:39]: Would be cool, if we could split this into multiple test cases that describe the case (e.g. `testCoundThreshWithNegativeNumbers` or using `@DisplayName`) so it's much easier to see what is actually failing.

Comment (TheAleWu) [reply w06h01/test/pgdp/CountTreshTest.java]: Good suggestion. I added it for KinguinSortTest as well ^^

Comment (nilsreichardt) [reply w06h01/test/pgdp/CountTreshTest.java]: Cool, thanks! Unfortunately, I have not started with W06 so I'm not able to make a complete review yet. Therefore, would be great when someone could approve the changes 👍

**Comment by Anatoly03** [comment]
Hab's genauer durchgeschaut, die Tests funktionieren ♡( •ॢ◡-ॢ)✧˖° ♡

**Comment by julian-hartl** [comment]
@TheAleWu Would be great if you could add tests that check if the next and prev nodes are correctly set after the kinguin sort.

**Comment by TheAleWu** [comment]
@julian-hartl that check should already be included by the usage of TestBase's assertListEquals - or did you mean something else?

**Review by MaximilianAnzinger — APPROVED** [review]

**Pull request merged**